nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/manager/ReportingTaskEntityMerger.java

```diff
@@ -29,6 +29,17 @@
 
 public class ReportingTaskEntityMerger implements ComponentEntityMerger<ReportingTaskEntity> {
 
+    @Override
+    public void merge(ReportingTaskEntity clientEntity, Map<NodeIdentifier, ReportingTaskEntity> entityMap) {
+        ComponentEntityMerger.super.merge(clientEntity, entityMap);
+        for (Map.Entry<NodeIdentifier, ReportingTaskEntity> entry : entityMap.entrySet()) {
+            final ReportingTaskEntity entityStatus = entry.getValue();
+            if (clientEntity != entityStatus) {
```

Comment:
This looks like it should use clientEntity.equals() instead of the instance comparison.

Reply:
I tested this out, I found that .equals() will likely return true because the object contents are very often the same, however the check was intended to skip only the clientEntity object that is inside the entityMap so we don't merge the clientEntity with itself and it uses the same reference in that map. So whilst the objects are the same, we don't want to skip over them in the case of merging their active threads for example.

Let me know if this sounds wrong.

Reply:
Thanks for the explanation, that makes sense. Given the `ReportingTaskEntity` does not override `equals()` to include a comparison of other fields, such as active threads, leaving this check as it is sounds good.